svelte/src/scripts/filter/index.ts

```diff
@@ -1,11 +1,37 @@
-import type { ConfigInterface, GraphData } from '../../types';
+import type { ConfigInterface, GraphData, GraphDataEdge, GraphDataNode } from '../../types';
 import { doCollapseNodes } from './collapse-nodes';
 import { liftDependencies } from './lift-edges';
 
 export function filter(config: ConfigInterface, graphData: GraphData) {
+	// reset all
+	// TODO: add the reasoning for this inside the doc.
+	resetNodeMemberToOriginal(graphData.flattenNodes);
+	resetLinksSourceAndTargetToOriginal(graphData.links);
+	// order doesn't matter here. TODO: add reasoning inside doc
 	// handle dependency lifting
-	graphData.links = liftDependencies(config, graphData);
-
+	liftDependencies(config);
 	// handle collapsed vertices
-	doCollapseNodes(config, graphData);
+	doCollapseNodes(config);
+}
+
+function doCombineEdgesWeight(config: ConfigInterface, graphData: GraphData) {
+	// TODO: impement this, and move it to another file
```

Comment:
combining edges weight was  was done inside lift edges and collapse node. I think we can move them here, in one place. not yet implemented. but should use the same function as previous. just pushing this here since I don't think it's gonna be blocking, and can be done parallel with other things, which means it can be done later.